rv-monitor-rt/src/main/java/com/runtimeverification/rvmonitor/java/rt/observable/UniqueMonitorTraceCollector.java

```diff
@@ -42,6 +42,7 @@ private String getFrequencyMap(Map<String, Integer> traceFrequencyMap) {
 
     private void analyzeUniqueTraces() {
         uniqueWriter.println("=== UNIQUE TRACE STATS ===");
+        if (traceDB.size() == 0) return; // prevent IndexOutOfBoundsException when TraceDB is empty
```

Comment:
maybe write in the `uniqueWriter` that the set of traces is empty before returning.

also, format code as:

```java
if (traceDB.size() == 0) {
   return;
}
```